### PR TITLE
Quicklink bugfix

### PR DESCRIPTION
**Pull request by daschuer**
This fixes https://bugs.launchpad.net/mixxx/+bug/1724222

### CODE REVIEW

**Comment by ronso0** [comment]
Bug fixed, tested on Ubuntu "Studio" 14.04 rt-kernel

**Comment by Be-ing** [comment]
Ready for merge?

File: src/library/sidebarmodel.cpp [hunk 1]
```diff
@@ -381,10 +383,12 @@ void SidebarModel::slotRowsRemoved(const QModelIndex& parent, int start, int end
     endRemoveRows();
 }
 
+void SidebarModel::slotModelAboutToBeReset() {
+    void beginResetModel();
```
Comment (uklotzde) [inline src/library/sidebarmodel.cpp]: Strayed 'void' from copy&paste, also in the following function.

File: src/library/browse/browsefeature.cpp [hunk 2]
```diff
@@ -133,7 +133,13 @@ void BrowseFeature::slotAddQuickLink() {
     QVariant vpath = m_pLastRightClickedItem->getData();
     QString spath = vpath.toString();
     QString name = extractNameFromPath(spath);
-    m_pQuickLinkItem->appendChild(name, vpath);
+
+    QModelIndex parent = m_childModel.index(m_pQuickLinkItem->parentRow(), 0);
+    auto pNewChild = new TreeItem(this, name, vpath);
```
Comment (uklotzde) [inline src/library/browse/browsefeature.cpp]: There is a very low probability that pNewChild item leaks until it has been inserted into the model. The key word 'auto' hides the plain pointer, so I would recommend at least to fix this by making it explicit.

Using a unique_ptr with get() when appending and calling release() after inserting would be fail safe.

**Comment by uklotzde** [comment]
Did not test it, but I found some minor coding issues that should be fixed.

**Comment by daschuer** [comment]
Notes addressed.

**Comment by Be-ing** [comment]
ping @uklotzde: ready for merge?

**Comment by uklotzde** [comment]
LGTM. Fixes the crash.

I still don't see any removable devices as also mentioned in the bug, but that's a different issue.

**Pull request merged**